### PR TITLE
feat(kuma-cp) e2e timeout test

### PR DESCRIPTION
**Pull request by lobkovilya**
### Summary

e2e test for Timeout policy

### CODE REVIEW

**Review by bartsmykla — APPROVED** [review]
lgtm

File: test/e2e/timeout/timeout_universal_test.go [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+			"curl", "-v", "--fail", "echo-server_kuma-test_svc_8080.mesh")
+		Expect(err).ToNot(HaveOccurred())
+		Expect(stdout).To(ContainSubstring("HTTP/1.1 200 OK"))
+
+		start := time.Now()
+		_, _, err = universalCluster.Exec("", "", "demo-client",
+			"curl", "-v", "--fail", "echo-server_kuma-test_svc_8080.mesh")
+		Expect(err).ToNot(HaveOccurred())
+		elapsed := time.Since(start)
+		fmt.Printf("Elapsed time with FaultInjection: %v\n", elapsed)
```
Comment (jakubdyszkiewicz) [inline test/e2e/timeout/timeout_universal_test.go:95]: is it needed?

File: test/e2e/timeout/timeout_universal_test.go [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+		err = NewClusterSetup().Install(YamlUniversal(timeout)).Setup(universalCluster)
+		Expect(err).ToNot(HaveOccurred())
+
+		// then
+		Eventually(func() bool {
+			start := time.Now()
+			_, _, _ = universalCluster.Exec("", "", "demo-client",
+				"curl", "-v", "--fail", "echo-server_kuma-test_svc_8080.mesh")
+			elapsed := time.Since(start)
+			return 2*time.Second <= elapsed && elapsed < 3*time.Second
```
Comment (jakubdyszkiewicz) [inline test/e2e/timeout/timeout_universal_test.go:109]: is there a response from Envoy that indicates that there was a timeout error?

File: test/e2e/timeout/timeout_universal_test.go (hunk 1 above)
Comment (lobkovilya) [reply test/e2e/timeout/timeout_universal_test.go]: Not really

File: test/e2e/timeout/timeout_universal_test.go (hunk 2 above)
Comment (lobkovilya) [reply test/e2e/timeout/timeout_universal_test.go]: Yes, thanks to @bartsmykla fix in https://github.com/kumahq/kuma/pull/1479, I figured out that `stdout` is not empty :)

**Review by jakubdyszkiewicz — APPROVED** [review]

**Pull request merged**